servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -0,0 +1,590 @@
+         * @param maxDelay The maximum amount of delay that will be introduced.
+         * @param maxRetries The maximum retries before it gives up.
+         * @param timerExecutor {@link Executor} to be used to schedule timers for backoff.
+         * It takes precedence over an alternative timer {@link Executor} from
+         * {@link #newStrategy(Executor)} argument
+         * @return A new retrying {@link BackOffPolicy} which adds an exponentially increasing
+         * delay between retries with jitter
+         */
+        public static BackOffPolicy ofExponentialBackoffDeltaJitter(
+                final Duration initialDelay, final Duration jitter, final Duration maxDelay, final int maxRetries,
+                final Executor timerExecutor) {
+            return new BackOffPolicy(initialDelay, jitter, maxDelay, timerExecutor, true, maxRetries);
+        }
+
+        /**
+         * Builds a new retry strategy {@link BiIntFunction} for retrying with
+         * {@link Publisher#retryWhen(BiIntFunction)}, {@link Single#retryWhen(BiIntFunction)}, and
+         * {@link Completable#retryWhen(BiIntFunction)} or in general with an alternative timer {@link Executor}.
+         *
+         * @param alternativeTimerExecutor {@link Executor} to be used to schedule timers for backoff if no executor
+         * was provided at the build time
+         * @return a new retry strategy {@link BiIntFunction}
+         */
+        public BiIntFunction<Throwable, Completable> newStrategy(final Executor alternativeTimerExecutor) {
+            if (initialDelay == null) {
+                return (count, throwable) -> count <= maxRetries ? completed() : failed(throwable);
+            } else {
+                final Executor effectiveExecutor = timerExecutor == null ?
+                        requireNonNull(alternativeTimerExecutor) : timerExecutor;
+                if (exponential) {
+                    assert maxDelay != null;
+                    return jitter == FULL_JITTER ?
+                            retryWithExponentialBackoffFullJitter(
+                                    maxRetries, t -> true, initialDelay, maxDelay, effectiveExecutor) :
+                            retryWithExponentialBackoffDeltaJitter(
+                                    maxRetries, t -> true, initialDelay, jitter, maxDelay, effectiveExecutor);
+                } else {
+                    return jitter == FULL_JITTER ?
+                            retryWithConstantBackoffFullJitter(
+                                    maxRetries, t -> true, initialDelay, effectiveExecutor) :
+                            retryWithConstantBackoffDeltaJitter(
+                                    maxRetries, t -> true, initialDelay, jitter, effectiveExecutor);
+                }
+            }
+        }
+    }
+
+    /**
+     * Default request retry policy builder.
+     */
+    public static final class DefaultRequestRetryPolicyBuilder {
```

Comment:
It's hard to discover that this class exists. Did you consider moving these methods to the filter builder? All boolean flags + user-provided `BiFunction` can be joined using logical "or".

Reply:
The reason this was introduced was that we have different APIs for retrying requests & responses. Having request flavored flags on the builder, where response retrying abilities are also provided is misleading.
I can advertise this class further on the request retrying function API. WDYT?